File: Backend/src/main/java/com/example/backend/entity/CategoryEntity.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;


//@Data – это сокращенная аннотация, сочетающая возможности @ToString,
// @EqualsAndHashCode, @Getter @Setter и @RequiredArgsConstructor

@Entity
@Table(name = "recipe_categories")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CategoryEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, unique = true, length = 50)
    private String name;

    @Column(length = 250)
    private String description;
}

File: MyApplication/data/src/main/java/com/example/data/Recipes/model/CategoryEntity.kt
package com.example.data.Recipes.model

import com.example.domain.Recipes.model.Category

data class CategoryEntity(
    val id: Int,
    val name: String,
    val description: String?
) {
    fun toDomain(): Category {
        return Category(
            id = id,
            name = name,
            description = description
        )
    }

}